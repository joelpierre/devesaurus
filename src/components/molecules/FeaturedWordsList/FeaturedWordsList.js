import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { graphql, Link, StaticQuery } from 'gatsby';
import styles from './FeaturedWordsList.module.scss';

export const PureFeaturedWordsList = ({ allWordpressWpWord }) => (
  <>
    {allWordpressWpWord.edges.map(({ node }) => {
      return (
        <li
          data-test="component-pure-featured-words-list"
          key={node.id}
          className={classNames(styles['featured-words-list__item'])}
        >
          <Link
            className={classNames(styles['featured-words-list__link'])}
            to={`/definition/${node.slug}`}
          >
            {node.title}
          </Link>
        </li>
      );
    })}
  </>
);

const FeaturedWordsList = ({ className }) => (
  <div
    data-test="component-featured-words-list"
    className={classNames(className, styles['featured-words-list'])}
  >
    <ul
      data-test="featured-words-list"
      className={classNames(styles['featured-words-list__list'])}
    >
      <StaticQuery
        data-test="component-featured-words-list-query"
        query={query}
        render={props => <PureFeaturedWordsList {...props} />}
      />
    </ul>
  </div>
);

FeaturedWordsList.defaultProps = {
  className: undefined,
};

FeaturedWordsList.propTypes = {
  className: PropTypes.string,
};

export default FeaturedWordsList;

const query = graphql`
  {
    allWordpressWpWord {
      edges {
        node {
          id
          title
          slug
        }
      }
    }
  }
`;
