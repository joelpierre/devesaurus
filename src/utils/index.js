export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
