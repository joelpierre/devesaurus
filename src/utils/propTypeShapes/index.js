import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  word: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  syllables: PropTypes.shape({
    count: PropTypes.number.isRequired,
    list: PropTypes.instanceOf(Array).isRequired,
  }).isRequired,
  pronunciation: PropTypes.string,
  results: PropTypes.arrayOf(PropTypes.shape({
    definition: PropTypes.string.isRequired,
    partOfSpeech: PropTypes.oneOf(['noun', 'verb', 'adjective', 'pronoun', 'adverb', 'preposition', 'conjunction', 'interjection']),
  })).isRequired,
  category: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  tags: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  })),
};
