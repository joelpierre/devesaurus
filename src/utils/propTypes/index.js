import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
