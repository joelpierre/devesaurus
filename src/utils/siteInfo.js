import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
