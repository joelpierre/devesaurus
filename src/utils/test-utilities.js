import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';

import rootReducer from '../store/reducers';
// import { sagaMiddleware } from '../store/configureStore';

// /**
//  * Store Factory
//  * @param initialState
//  * @returns {Store<any, Action> & *}
//  */
// export const storeFactory = (initialState) => {
//   const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
//   return createStoreWithMiddleware(rootReducer, initialState);
// };


/**
 * Return node(s) with the given data-test attribute.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper.
 * @param {string} val - Value of the data-test attribute.
 * @returns {ShallowWrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


/**
 * Check the props for a component
 * @param component
 * @param conformingProps
 */
export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name,
  );
  expect(propError)
    .toBeUndefined();
};
