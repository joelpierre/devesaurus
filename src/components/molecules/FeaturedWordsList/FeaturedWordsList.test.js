import React from 'react';
import { shallow } from 'enzyme';

import { checkProps, findByTestAttr, matchSnapshot } from '../../../utils/test';

import FeaturedWordsList, { PureFeaturedWordsList } from './FeaturedWordsList';

const defaultProps = {};

const defaultPureProps = {
  allWordpressWpWord: {
    edges: [
      {
        node: {
          id: 1,
          title: 'title',
          slug: 'slug',
          acf: {
            pronunciation: 'boom',
            origin: {
              value: 'UK',
              label: 'GB',
            },
          },
          word_tags: {
            id: 1,
            slug: 'tag-slug',
            name: 'tag-name',
            taxonomy: 'word_tag',
          },
          word_cats: {
            id: 1,
            slug: 'category-slug',
            name: 'category-name',
            taxonomy: 'word_category',
          },
        },
      },
    ],
  },
};

/**
 * Factory function to create a ShallowWrapper for the FeaturedWordsList component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<FeaturedWordsList {...setupProps} />);
};

const pureSetup = (props = {}) => {
  const setupProps = { ...defaultPureProps, ...props };
  return shallow(<PureFeaturedWordsList {...setupProps} />);
};

describe('<PureFeaturedWordsList/>', () => {
  let wrapper;
  let pureWrapper;

  beforeEach(() => {
    wrapper = setup();
    pureWrapper = pureSetup();
  });

  it('renders the FeaturedWordsList Component without errors query', () => {
    const component = findByTestAttr(
      wrapper,
      'component-featured-words-list-query'
    );
    expect(component.length).toBe(1);
  });

  it('renders the FeaturedWordsList Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-featured-words-list');
    expect(component.length).toBe(1);
  });

  it('renders the FeaturedWordsList list without errors', () => {
    const component = findByTestAttr(wrapper, 'featured-words-list');
    expect(component.length).toBe(1);
  });

  it('renders the PureFeaturedWordsList Component without errors', () => {
    const component = findByTestAttr(
      pureWrapper,
      'component-pure-featured-words-list'
    );

    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(FeaturedWordsList, defaultProps);
  });

  it('should render with correct pureProps', () => {
    checkProps(PureFeaturedWordsList, defaultPureProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should match pure snapshot', () => {
    matchSnapshot(pureWrapper);
  });
});
