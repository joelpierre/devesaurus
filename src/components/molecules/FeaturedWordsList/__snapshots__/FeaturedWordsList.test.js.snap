// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PureFeaturedWordsList/> should match pure snapshot 1`] = `
<Fragment>
  <li
    className="featured-words-list__item"
    data-test="component-pure-featured-words-list"
    key="1"
  >
    <mockConstructor
      className="featured-words-list__link"
      to="/word/slug"
    >
      title
    </mockConstructor>
  </li>
</Fragment>
`;

exports[`<PureFeaturedWordsList/> should match snapshot 1`] = `
<div
  className="featured-words-list"
  data-test="component-featured-words-list"
>
  <ul
    className="featured-words-list__list"
    data-test="featured-words-list"
  >
    <mockConstructor
      data-test="component-featured-words-list-query"
      render={[Function]}
    />
  </ul>
</div>
`;
